modules/FIC_Github.py

```diff
@@ -114,3 +120,172 @@ def push(self):
+        if self.build_puppet_version == self.release_version:
+            if self.release_version != self.local_version:
+                return True
+            else:
+                return False
+        else:
+            return False
+
+    def _last_checked(self):
+        import datetime
+        self.last_check = datetime.datetime.strptime(json.load(self.load(CHANGELOG_REPO_PATH, self.repo_name.lower() + ".json")).get("0").get("last_checked"), "%Y-%m-%d %H:%M:%S")
+
+    def _commit_iterator(self):
+        self.commit_number = 0
+        for current_commit in self.repo_data.commits(since=self.last_check):
+            self._commit_filter(current_commit)
+        self.keyword = None
+
+    def _store_data(self, current_commit):
+        self._get_sha(current_commit)
+        self._get_message(current_commit)
+        self._get_date(current_commit)
+        self._get_author(current_commit)
+        self._get_author_email(current_commit)
+        self._get_url(current_commit)
+        self._get_files()
+
+    def _get_sha(self, commit):
+        self.commit_sha = commit.sha
+
+    def _get_message(self, commit):
+        self.commit_message = commit.message
+
+    def _get_date(self, commit):
+        self.commit_date = commit.commit.author.get("date")
+
+    def _get_author(self, commit):
+        self.commit_author = commit.commit.author.get("name")
+
+    def _get_author_email(self, commit):
+        self.commit_author_email = commit.commit.author.get("email")
+
+    def _get_url(self, commit):
+        self.commit_url = commit.url
+
+    def _get_files(self):
+        self.commit_files_changed = []
+        for item in (range(len(self.repo_data.commit(sha=self.commit_sha).files))):
+            self.commit_files_changed.append(self.repo_data.commit(sha=self.commit_sha).files[item].get('filename'))
+
+    def _compare_files(self):
+        for folder_to_check in range(len(self.folders_to_check)):
+            for changed_folder in range(len(self.commit_files_changed)):
+                if str(self.folders_to_check[folder_to_check]) in str(self.commit_files_changed[changed_folder]):
+                    return True
+
+    def _construct_commit(self):
+        self.list_of_commits.update({self.commit_number: {'sha': self.commit_sha,
+                                                          'url': self.commit_url,
+                                                          'author': self.commit_author,
+                                                          'author_email': self.commit_author_email,
+                                                          'message': self.commit_message,
+                                                          'date': self.commit_date,
+                                                          'files': self.commit_files_changed}})
+
+    def _commit_filter(self, commit_content):
+        if self.repo_type == "commit-keyword":
+            if self.keyword in self.commit_message:
```

Comment:
those can be combined, something like:
```Python
if (self.repo_type == "commit-keyword") and (self.keyword in self.commit_message)
   self.commit_number += 1
   self._store_data(commit_content)
   self._construct_commit()
```

Reply:
It was done in: https://github.com/mozilla-releng/firefox-infra-changelog/pull/448/commits/16f80b00b7463d786334be5bc955e8dfac03b020